pkg/node/local_node_store.go

```diff
@@ -72,113 +46,75 @@ type LocalNodeStoreParams struct {
 	Logger    *slog.Logger
 	Lifecycle cell.Lifecycle
 	Sync      LocalNodeSynchronizer `optional:"true"`
```

Comment:
Just wondering. How big of a pain would it be dropping optional from here, just to prevent surprises? I assume that was/is needed for tests only, as otherwise the LocalNodeSynchronizer is always provided.